common/src/main/java/com/datastrato/gravitino/dto/requests/FilesetCreateRequest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.dto.requests;
+
+import com.datastrato.gravitino.file.Fileset;
+import com.datastrato.gravitino.rest.RESTRequest;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.extern.jackson.Jacksonized;
+import org.apache.commons.lang3.StringUtils;
+
+@Getter
+@EqualsAndHashCode
+@ToString
+@Builder
+@Jacksonized
+@NoArgsConstructor
+@AllArgsConstructor
+public class FilesetCreateRequest implements RESTRequest {
+
+  @JsonProperty("name")
+  private String name;
+
+  @Nullable
+  @JsonProperty("comment")
+  private String comment;
+
+  @Nullable
+  @JsonProperty("type")
+  private Fileset.Type type;
```

Comment:
If this Is null, we'd better provide a default value. Otherwise, there will be NPE in HadoopCatalog.

Reply:
I give a default value in FilesetDTO builder: https://github.com/datastrato/gravitino/pull/1795/files#diff-1d5e20c0e26e4ddafcf9209a1ab3edc7045af6e8e4090ccfd343e6fa20ba5940R87

Is it available?

Reply:
I see, thanks.

Reply:
I don't think having a default value in `FilesetDTO` can solve this problem. The variable here will be used for `createFileset()` method. And we don't have the check here, so it will possibly an issue.

Reply:
Sorry for not saying it clearly. IMO, I think it is OK to have null value for `type` in REST protocol, but we'd better have a default value when using this `type`.

Reply:
yes, I know. I remove the default from the REST protocol to keep the original request for better audit.